migration.rst

```diff
@@ -0,0 +1,104 @@
+.. index::
+   single: Migration
+
+Migrating an existing application to Symfony
+============================================
+
+When you have an existing application that was not built with Symfony,
+you might want to move over parts of that application without rewriting
+the existing logic completely. For those cases there is a pattern called
+`Strangler Application`_. The basic idea of this pattern is to create a
+new application that gradually takes over functionality from an existing
+application. This migration approach can be implemented with Symfony in
+various ways and has some benefits over a rewrite such as being able
+to introduce new features in the existing application and reducing risk
+by avoiding a "big bang"-release for the new application.
+
+Prerequisites
+-------------
+
+Before you start introducing Symfony to the existing application, you
+have to ensure certain requirements are met by your existing application.
+First of all your environment needs to be able to support the minimum
+requirements for both applications. In other words, when Symfony requires
+PHP 7.1 and your existing application does not support this PHP version,
+you will probably have to upgrade your legacy project. You can find out
+the :doc:`requirements </reference/requirements>` for running Symfony and
+compare them with your current application's environment to make sure you
+are able to run both applications on the same system.
+
```

Comment:
TODO: Add paragraph on which target version to choose (LTS vs. most recent release or a current stable release)